boot/zephyr/CMakeLists.txt

```diff
@@ -290,6 +290,42 @@ if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
     DEPENDS ${KEY_FILE}
     )
   zephyr_library_sources(${GENERATED_PUBKEY})
+
+
+endif()
+
+if(CONFIG_BOOT_USE_ENCRYPT_KEY_FILE)
+  if(NOT CONFIG_BOOT_ENCRYPT_KEY_FILE STREQUAL "")
+    if(IS_ABSOLUTE ${CONFIG_BOOT_ENCRYPT_KEY_FILE})
+      set(ENCRYPTION_KEY_FILE ${CONFIG_BOOT_ENCRYPT_KEY_FILE})
+    elseif((DEFINED CONF_DIR) AND
+    (EXISTS ${CONF_DIR}/${CONFIG_BOOT_ENCRYPT_KEY_FILE}))
+      set(ENCRYPTION_KEY_FILE ${CONF_DIR}/${CONFIG_BOOT_ENCRYPT_KEY_FILE})
+    else()
+      set(ENCRYPTION_KEY_FILE ${MCUBOOT_DIR}/${CONFIG_BOOT_ENCRYPT_KEY_FILE})
+    endif()
+    message("MCUBoot bootloader encryption key file: ${ENCRYPTION_KEY_FILE}")
+
+    if(NOT EXISTS "${ENCRYPTION_KEY_FILE}")
+      message(FATAL_ERROR "Cannot find encryption key file: ${ENCRYPTION_KEY_FILE}")
+    endif()
+
+    set(GENERATED_PRIVKEY ${ZEPHYR_BINARY_DIR}/autogen-privkey.c)
+    add_custom_command(
+        OUTPUT ${GENERATED_PRIVKEY}
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${MCUBOOT_DIR}/scripts/imgtool.py
+      getpriv
+      -k
+      ${ENCRYPTION_KEY_FILE}
```

Comment:
You could consider adding the --minimal option.
Which was used for the default ec256 and x25519 keys.

Reply:
Added `--minimal` for ec256 and x25519 keys only, and not for rsa key. The generated keys are equal to the keys removed from `keys.c`.